source/opt/ir_context.cpp

```diff
@@ -83,7 +83,17 @@ void IRContext::KillInst(ir::Instruction* inst) {
     }
   }
 
-  inst->ToNop();
+  Instruction* next_instruction = nullptr;
+  if (inst->IsInAList()) {
+    next_instruction = inst->NextNode();
+    inst->RemoveFromList();
+    delete inst;
+  } else {
+    // Needed for instructions that are not part of a list like OpLabels,
+    // OpFuntion, OpFunctionEnd, etc..
```

Comment:
s/OpFuntion/OpFunction/

Reply:
I don't understand, aren't all instructions in various lists in ir::Module?

Reply:
No, if you look at the definition of a BasicBlock the label is a single instruction in the basic block.  Then we have a list to hold the other instructions.

OpFunction and OpFunctionEnd are similar.

Reply:
Ah, yes.  I figured I was missing something obvious.  Thanks.